grease/src/Grease/ErrorDescription.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Grease.ErrorDescription (ErrorDescription (..), concretizeErrorDescription, ppMacawError, ppErrorDesc) where
+
+import Data.List qualified as List
+import Data.Macaw.Symbolic.Memory (MacawError (UnmappedGlobalMemoryAccess))
+import Data.Macaw.Symbolic.Memory qualified as MSM
+import Lang.Crucible.LLVM.Errors qualified as CLLVM
+import Lang.Crucible.LLVM.Errors qualified as Mem
+import Lang.Crucible.LLVM.MemModel qualified as CLLVM
+import Lang.Crucible.LLVM.MemModel.CallStack qualified as LLCS
+import Lang.Crucible.LLVM.MemModel.CallStack qualified as Mem
+import Lang.Crucible.LLVM.MemModel.Pointer qualified as Mem
+import Prettyprinter qualified as PP
+import What4.Expr qualified as W4
+import What4.Interface qualified as W4
+import What4.Interface qualified as WI
+
+concretizeErrorDescription ::
+  (WI.IsExprBuilder sym, sym ~ W4.ExprBuilder t st fs) =>
+  sym ->
+  W4.GroundEvalFn t ->
+  ErrorDescription sym ->
+  IO (ErrorDescription sym)
+concretizeErrorDescription sym (W4.GroundEvalFn gFn) (CrucibleLLVMError bb cs) = do
+  bb' <- Mem.concBadBehavior sym gFn bb
+  pure (CrucibleLLVMError bb' cs)
+concretizeErrorDescription sym (W4.GroundEvalFn gFn) (MacawMemError memerr) = do
+  (UnmappedGlobalMemoryAccess ptrVal) <- pure memerr
+  cptr <- Mem.concPtr sym gFn ptrVal
+  pure $ MacawMemError (UnmappedGlobalMemoryAccess cptr)
+
+-- | An error either from the underlying LLVM memory model or
+-- from Macaw.
+data ErrorDescription sym
+  = CrucibleLLVMError (CLLVM.BadBehavior sym) LLCS.CallStack
+  | MacawMemError (MSM.MacawError sym)
+
+ppDelimitedObj :: [PP.Doc a] -> PP.Doc a
+ppDelimitedObj lst = PP.sep $ "(" : lst ++ [")"]
+
+-- TODO(#310) Pretty print macaw error using a macaw implementation
+ppMacawError :: W4.IsExpr (W4.SymExpr sym) => MSM.MacawError sym -> PP.Doc a
+ppMacawError (MSM.UnmappedGlobalMemoryAccess ptrVal) = PP.sep ["(", "UnmappedGlobalMemoryAccess", CLLVM.ppPtr ptrVal, ")"]
```

Comment:
Generally, `Doc`s are for showing to users. Can we describe this error in a way that's more accessible to users? Perhaps

```suggestion
ppMacawError (MSM.UnmappedGlobalMemoryAccess ptrVal) =
  "Read or write of an unmapped pointer:" PP.<+> CLLVM.ppPtr ptrVal
```